### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by konradweiss**
Here we add a table that communicates the state and locations of all language frontends.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,14 +116,34 @@ val translationConfig = TranslationConfiguration
...
+| TypeScript/JavaScript | cpg-language-typescript | [main](https://github.com/Fraunhofer-AISEC/cpg) | `developing` |
+| Ruby | cpg-language-ruby | [main](https://github.com/Fraunhofer-AISEC/cpg) | `developing` |
+| Python-Quantum | T.B.A | T.B.A | `experimental` |
+
+### Languages and Configuration
+`cpg-core` contains the graph nodes, language-independent passes that add semantics to the cpg-AST. Languages are developed in separate gradle submodules. 
+To include the desired language submodules, simply toggle them on in your local `gradle.properties` file by setting the properties to `true`, e.g., (`enableGoFrontend=true`).
 We provide a sample file with all languages switched on [here](./gradle.properties.example).
-Instead of manually editing the `gradle.properties` file, you can also use the `configure_frontends.sh` script, which edits the properties for you.
+Instead of manually editing the `gradle.properties` file, you can also use the `configure_frontends.sh` script, which edits the properties for you. Some languages, such as Golang, depend on other native libraries that have to be installed manually.
```
Comment (oxisto) [inline README.md:146]: Go does not need any manual installation (other than Go itself).

File: README.md [hunk 2]
```diff
@@ -116,14 +116,34 @@ val translationConfig = TranslationConfiguration
...
-
-Some languages, such as Golang are experimental and depend on other native libraries. Therefore, they are not included in the `cpg-core` module but have separate gradle submodules.
-To include the desired submodules simply toggle them on in your local `gradle.properties` file by setting the value of the properties to `true` e.g., (`enableGoFrontend=true`).
+## Development
+This section describes languages, how well they are supported, and how to use and develop them yourself.
+
+### Language Support
+Languages are maintained to different degrees, and are noted in the table below with:
+- `maintained`: if they are mostly feature complete and bugs have priority of being fixed.
+- `developing`: if the language is currently being worked on to reach a state of feature completeness.
```
Comment (oxisto) [inline README.md:125]: I think we used the term “incubating”

File: README.md [hunk 3]
```diff
@@ -116,14 +116,34 @@ val translationConfig = TranslationConfiguration
...
+  
+The current state of languages is:
+
+| Language | Module | Branch | State |
+|---|---|---|---|
+| Java | cpg-language-java | [main](https://github.com/Fraunhofer-AISEC/cpg) | `maintained` |
+| C++ | cpg-language-cxx | [main](https://github.com/Fraunhofer-AISEC/cpg) | `maintained` |
+| Python | cpg-language-python | [main](https://github.com/Fraunhofer-AISEC/cpg) | `maintained` |
+| Go | cpg-language-go | [main](https://github.com/Fraunhofer-AISEC/cpg) | `maintained` |
+| LLVM | cpg-language-llvm | [main](https://github.com/Fraunhofer-AISEC/cpg) | `maintained` |
```
Comment (oxisto) [inline README.md:137]: I would consider this more as incubating

File: README.md [hunk 4]
```diff
@@ -116,14 +116,34 @@ val translationConfig = TranslationConfiguration
...
+| TypeScript/JavaScript | cpg-language-typescript | [main](https://github.com/Fraunhofer-AISEC/cpg) | `developing` |
```
Comment (oxisto) [inline README.md:138]: Experimental

File: README.md [hunk 5]
```diff
@@ -116,14 +116,34 @@ val translationConfig = TranslationConfiguration
...
+| Ruby | cpg-language-ruby | [main](https://github.com/Fraunhofer-AISEC/cpg) | `developing` |
```
Comment (oxisto) [inline README.md:139]: Experimental

File: README.md (hunk 2 above)
Comment (konradweiss) [reply README.md:125]: Yes I changed It cause we were not sure what was a good idea or not. I chose this terms due to them being more commonly used, but they might miss a notion that I was not aware we wanted to go for. What do you think, leave it or change `developing` to `incubating` and/or `experimental` to `research`?

File: README.md (hunk 3 above)
Comment (konradweiss) [reply README.md]: I hope I changed the correct line. Here the lowest, namely LLVM.

File: README.md (hunk 2 above)
Comment (oxisto) [reply README.md:125]: `experimental` seems fine for me. Personally, I have not heard `developing` for such projects, but `incubating` seems a common term, such as used by the CNCF https://www.cncf.io/projects/, Apache https://incubator.apache.org/projects/, Eclipse https://www.eclipse.org/projects/dev_process/#6_2_3_Incubation

**Review by oxisto — APPROVED** [review]

**Pull request merged**